src/AstAnalyses.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved.
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AstAnalyses.cpp
+ *
+ * Implementation of AST analyses classes
+ *
+ ***********************************************************************/
+
+#include <cassert>
+#include <utility>
+
+#include "AstAnalyses.h"
+#include "AstProgram.h"
+#include "AstUtils.h"
+
+namespace souffle {
+
+const size_t AuxiliaryArity::getEvaluationArity(const AstAtom* atom) const {
+    if (atom->getName().getName().rfind("@delta_") == 0) {
```

Comment:
Do these need to use `rfind` or would `find` do?

Reply:
Ideally, these relations are marked differently and we don't discover their character via the name.

Reply:
However, this is legacy code that will be tidied later with Yun's info relation representation.